### PR TITLE
gh-96765 Add docs order of configparser.ConfigParser.read() arguments

### LINKED ISSUE #96765: Order of configparser.ConfigParser.read() arguments
# Bug report

When multiple INI configuration files with identical section(s)  and key(s) are fed to [`configparser.ConfigParser.read`](https://docs.python.org/3/library/configparser.html#configparser.ConfigParser.read), the result is undefined / not documented. Actually the last file containing a given section/key takes precedence over previous files, as I would have expected from a straightforward implementation. However:
* This behaviour should be documented.
* ~~Perhaps this behaviour should be changed, and the first file take precedence over subsequent files. In practice that would break compatibility, but since the result is currently undocumented, that may be OK.~~

## Example

Here are two INI files:

**`1.ini`**
```ini
[section]
a = 1
b = "11"
```
**`2.ini`**
```ini
[section]
a = 2
b = "22"
```

And here is the order of the filenames changes the result:
```python
>>> import sys
>>> import configparser
>>> config = configparser.ConfigParser()
>>> 
>>> config.read(["1.ini", "2.ini"])
['1.ini', '2.ini']
>>> config.write(sys.stdout)
[section]
a = 2
b = "22"

>>> 
>>> config.read(["2.ini", "1.ini"])
['2.ini', '1.ini']
>>> config.write(sys.stdout)
[section]
a = 1
b = "11"

>>> 
```
# Your environment

- CPython versions tested on: 3.6.9
- Operating system and architecture: Ubuntu 18.04

### Linked PRs
* gh-107683
* gh-121664
* gh-121687
* gh-121688

### PR DESCRIPTION
**Pull request by xncbf**
Closes #96765

* Issue: gh-96765

----
:books: Documentation preview :books:: https://cpython-previews--107683.org.readthedocs.build/

### CODE REVIEW

**Comment by xncbf** [comment]
I think all versions 3.2 and above should apply.
Can someone help me with a backport?

**Comment by xncbf** [comment]
@ambv Could you please review this PR?

File: Doc/library/configparser.rst [hunk 1]
```diff
@@ -1043,6 +1043,27 @@ ConfigParser Objects
...
 
+      It is possible to read several configurations into a single
+      :class:`ConfigParser`, where the most recently added configuration has the
+      highest priority. Any conflicting keys are taken from the more recent
+      configuration while the previously existing keys are retained.
+
+      .. doctest::
+
+         >>> another_config = configparser.ConfigParser()
+         >>> another_config.read('example.ini')
```
Comment (vadmium) [inline Doc/library/configparser.rst:1054]: There is only one _read_ call, with a single file, so this does not follow the text you added, and seems out of place as an illustration.

**Comment by ambv** [comment]
Closed in favor of gh-121664.

**Pull request closed without merging**